pkg/sriov/operatorconfig.go

```diff
@@ -0,0 +1,146 @@
+package sriov
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+
+	srIovV1 "github.com/k8snetworkplumbingwg/sriov-network-operator/api/v1"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	sriovOperatorConfigName = "default"
+)
+
+// OperatorConfigBuilder provides a struct for SriovOperatorConfig object from the cluster and
+// a SriovOperatorConfig definition.
+type OperatorConfigBuilder struct {
+	// SriovOperatorConfig definition, used to create the SriovOperatorConfig object.
+	Definition *srIovV1.SriovOperatorConfig
+	// Created SriovOperatorConfig object.
+	Object *srIovV1.SriovOperatorConfig
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+	errorMsg  string
+}
+
+// PullOperatorConfig loads an existing SriovOperatorConfig into OperatorConfigBuilder struct.
+func PullOperatorConfig(apiClient *clients.Settings, nsname string) (*OperatorConfigBuilder, error) {
```

Comment:
Wonder if there can be different operator configs. In that case this should be `PullDefaultOperatorConfig`

And in case there can be different operator configs, that can be accessed with `PullOperatorConfigByName`.

Reply:
I don't think we can, it's a defined CRD: srIovV1.SriovOperatorConfig